### PR TITLE
Login oauth workaround

### PR DESCRIPTION
**Pull request by adnanhemani**
Created a new workflow for users of Google OAuth on Pivotal Tracker.

They must put in their API Token manually into the app from the login screen, and then Google OAuth will work for them in our app.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -55,6 +55,30 @@ def salted(username)
...
+    def set_new_api_key(params)
+      this_user = User.find_by_username(params[:username])
+      if not this_user
+        this_user = User.new(
+          username: params[:username],
+          password: (0...8).map { (65 + rand(26)).chr }.join,
+          token:    params[:api_key]
+        )
+      else
+        this_user.token = params[:api_key]
```
Comment (an-ju) [inline app/models/user.rb:68]: What was user.token used for?

**Comment by an-ju** [comment]
I think the solution looks fine.  Although I'd like to explain a bit more to students in the "set API key" page that we need to store their API keys in this app.

**Comment by an-ju** [comment]
Also, it would be good if students can remove their stored keys.  I think previously API keys were not stored in this app.

File: app/models/user.rb (hunk 1 above)
Comment (adnanhemani) [reply app/models/user.rb:68]: This is used to make all the calls to the Pivotal Tracker API, without it, we can't make any calls.

**Comment by adnanhemani** [comment]
Previously, we would ask students to log in with their credentials and then manually go find the API keys and store them. The issue here is that since students need to log in via OAuth, we're unable to hit the Pivotal Tracker API to find their API keys. That's why we need them to manually input them into the app.

In terms of the workflow, it's nothing different than before regarding the API tokens. We stored them before and we're still storing them now - albeit now we're asking for them directly rather than finding it through the Pivotal Tracker API.

**Comment by an-ju** [comment]
That sounds good to me.  Go ahead and merge the PR.

**Pull request merged**